persistence/rocksdb/src/main/java/org/infinispan/persistence/rocksdb/RocksDBStore.java

```diff
@@ -741,36 +741,32 @@ RocksDB open(Path location, DBOptions options) throws RocksDBException {
             boolean destroyDatabase = false;
             try (ReadOptions readOptions = new ReadOptions().setFillCache(false)) {
                RocksIterator optionalIterator = wrapIterator(db, readOptions, -1);
-               if (optionalIterator != null && (configuration.clearThreshold() > 0 || segments == null)) {
-                  try (RocksIterator it = optionalIterator) {
-                     for (it.seekToFirst(); it.isValid(); it.next()) {
-                        byte[] keyBytes = it.key();
-                        if (segments != null) {
-                           Object key = unmarshall(keyBytes);
-                           int segment = keyPartitioner.getSegment(key);
-                           if (segments.contains(segment)) {
-                              db.delete(defaultColumnFamilyHandle, keyBytes);
-                           }
-                        } else {
-                           db.delete(defaultColumnFamilyHandle, keyBytes);
-                           count++;
-
-                           if (count > configuration.clearThreshold()) {
-                              destroyDatabase = true;
-                              break;
+               if (optionalIterator != null) {
+                  if (segments == null) {
+                     deleteRange(defaultColumnFamilyHandle, optionalIterator);
+                  } else {
+                     if (configuration.clearThreshold() > 0) {
```

Comment:
This cannot use the `clearThreshold` here. You can end up deleting the db could include entries that aren't in the `IntSet`. All we can do is iterate over the entries.